app/src/main/java/org/p2p/wallet/moonpay/interactor/MoonpayBuyInteractor.kt

```diff
@@ -0,0 +1,122 @@
+package org.p2p.wallet.moonpay.interactor
+
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.p2p.wallet.home.model.Token
+import org.p2p.wallet.infrastructure.dispatchers.CoroutineDispatchers
+import org.p2p.wallet.infrastructure.network.data.ErrorCode
+import org.p2p.wallet.infrastructure.network.data.ServerException
+import org.p2p.wallet.moonpay.api.MoonpayBuyCurrencyResponse
+import org.p2p.wallet.moonpay.model.MoonpayBuyQuote
+import org.p2p.wallet.moonpay.model.MoonpayBuyResult
+import org.p2p.wallet.moonpay.repository.MoonpayApiMapper
+import org.p2p.wallet.moonpay.repository.NewMoonpayRepository
+import org.p2p.wallet.utils.isLessThan
+import java.math.BigDecimal
+
+const val DEFAULT_MIN_AMOUNT = 40
+
+private const val DEFAULT_AMOUNT = "1"
+private const val DEFAULT_PAYMENT_TYPE = "credit_debit_card"
+private val FIAT_CURRENCY_CODES = listOf("eur", "usd", "gbp")
+
+class MoonpayBuyInteractor(
```

Comment:
are you sure it's interactor responsibility to put dispatchers here? we put them in a repository

Reply:
I saw different approaches in our code. For example, `DeviceTokenRemoteRepository` doesn't use any dispatchers inside for going to api. If it is a rule about that I will refactor it and be aware